extensions/ql-vscode/src/databaseFetcher.ts

```diff
@@ -218,3 +322,72 @@ async function findDirWithFile(dir: string, ...toFind: string[]): Promise<string
   }
   return;
 }
+
+function looksLikeLgtmUrl(lgtmUrl: string | undefined): lgtmUrl is string {
+  if (!lgtmUrl) {
+    return false;
+  }
+
+  try {
+    const uri = Uri.parse(lgtmUrl, true);
+    if (uri.scheme !== "https") {
+      return false;
+    }
+
+    if (uri.authority !== "lgtm.com" && uri.authority !== "www.lgtm.com") {
+      return false;
+    }
+
+    const paths = uri.path.split("/").filter((segment) => segment);
+    return paths.length === 4 && paths[0] === "projects" && paths[1] === "g";
```

Comment:
```suggestion
    return paths.length >= 4 && paths[0] === "projects";
```
The URL pattern is `https://lgtm.com/projects/{provider}/{org}/{name}` . In addition to GitHub.com(`g`), LGTM currently hosts projects from Bitbucket (`b`), GitLab (`gl`) and plain git (`git`). 

After the `{provider}/{org}/{name}` path components, there may be the components related to sub pages. I think it would be best to accept those too.

Reply:
Makes sense.